### PR TITLE
iOS Support

### PR DESCRIPTION
**Pull request by BrentDaMage**
Remaking this pull-request since the last one was closed by iProgramMC accidentally.

### CODE REVIEW

**Comment by BrentDaMage** [comment]
Ready for review

File: source/world/level/path/BinaryHeap.cpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ void BinaryHeap::downHeap(int num)
 		float var9;
 		if (var5 >= m_count) {
 			var8 = nullptr;
-			var9 = INFINITY;
+			var9 = std::numeric_limits<float>::infinity();
```
Comment (iProgramMC) [inline source/world/level/path/BinaryHeap.cpp:74]: Why was this changed?

Comment (iProgramMC) [inline source/world/level/path/BinaryHeap.cpp:74]: Why was this changed?

File: source/client/model/HumanoidModel.cpp [hunk 2]
```diff
@@ -18,7 +18,9 @@ HumanoidModel::HumanoidModel(float a, float b):
 	m_leg1(0,  16),
 	m_leg2(0,  16)
 {
+	field_20 = false;
 	field_234 = 0;
+	field_20 = false;
```
Comment (iProgramMC) [inline source/client/model/HumanoidModel.cpp:23]: Remove the extraneous assignment

File: platforms/ios/minecraftpeAppDelegate.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
+//
+//  minecraftpeAppDelegate.h
+//  minecraftpe
+//
+//  Created by Brent on 10/12/23.
+//  Copyright (c) 2023 __MyCompanyName__. All rights reserved.
```
Comment (iProgramMC) [inline platforms/ios/minecraftpeAppDelegate.h:6]: Write "ReMinecraftPE" instead of `__MyCompanyName__` I guess.

File: source/world/level/path/BinaryHeap.cpp (hunk 1 above)
Comment (iProgramMC) [reply source/world/level/path/BinaryHeap.cpp:74]: I know, it's because `INFINITY` was added shockingly late, in C++11.

Comment (BrentDaMage) [reply source/world/level/path/BinaryHeap.cpp:74]: In classic C++ fashion...

File: source/client/model/HumanoidModel.cpp (hunk 2 above)
Comment (BrentDaMage) [reply source/client/model/HumanoidModel.cpp:23]: My bad, I moved that over from the header file without realizing it was already assigned at the top. Good catch.
